### ISSUE TITLE
MD035 should ignore the leading `>` when checking a pattern

### ISSUE THREAD

**Issue by nschonni** [DavidAnson/markdownlint#484]
When checking a document for a HR of `---` using
```json
  "MD035": {
    "style": "---"
  },
```
on the following
```md
> **Note:** _Remove this whole explanatory note before publishing_
>
> ---
>
> **Page front matter:**
>
> The frontmatter at the top of the page is used to define "page metadata".
> The values should be updated appropriately for the constructor.
```

It flags the `> ---` instead of ignoring the leading `> ` because it is in a blockquote

`[Expected: ---; Actual: > ---]`

**Comment by DavidAnson** [comment]
I think this is fixed in the latest version of the library for #430: https://dlaa.me/markdownlint/#%25m%23%20Issue%20484%0A%0AText%0A%0A---%0A%0A%3E%20Text%0A%3E%0A%3E%20---%0A%3E%0A%3E%20Text%0A

**Comment by nschonni** [comment]
OK, feel free to close. The only part I'm not sure the live site can test is if it was an issue with the explicit config causing the issue vs. the default "consistent"

**Comment by DavidAnson** [comment]
It should not matter, but you can include explicit config on the demo site with markdownlint-configure-file if you want to be sure.
